app/src/main/java/org/sopt/and/feature/signup/SignUpScreen.kt

```diff
@@ -16,72 +10,39 @@ import androidx.compose.foundation.layout.height
 import androidx.compose.foundation.layout.padding
 import androidx.compose.material3.Button
 import androidx.compose.material3.ButtonDefaults
-import androidx.compose.material3.Scaffold
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.remember
-import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.colorResource
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
-import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavHostController
 import org.sopt.and.R
 import org.sopt.and.component.SignUpTopBar
 import org.sopt.and.component.WavveCustomTextField
-import org.sopt.and.ui.theme.ANDANDROIDTheme
-import org.sopt.and.utils.KeyStorage.EMAIL
 import org.sopt.and.utils.KeyStorage.EMAIL_PATTERN
-import org.sopt.and.utils.KeyStorage.PASSWORD
 import org.sopt.and.utils.KeyStorage.PASSWORD_MAX_LENGTH
 import org.sopt.and.utils.KeyStorage.PASSWORD_MIN_LENGTH
 import org.sopt.and.utils.KeyStorage.PASSWORD_PATTERN
 import org.sopt.and.utils.toast
 import java.util.regex.Pattern
 
-class SignUpActivity : ComponentActivity() {
-
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        enableEdgeToEdge()
-        setContent {
-            ANDANDROIDTheme {
-                Scaffold(modifier = Modifier.fillMaxSize()) { innerPadding ->
-                    SignUp(
-                        modifier = Modifier.padding(innerPadding),
-                        onSignUpSuccess = { email, password ->
-                            handleSignUp(email, password)
-                        }
-                    )
-                }
-            }
-        }
-    }
-
-    private fun handleSignUp(email: String, password: String) {
-        val intent = Intent().apply {
-            putExtra(EMAIL, email)
-            putExtra(PASSWORD, password)
-        }
-        setResult(Activity.RESULT_OK, intent)
-        finish()
-    }
-}
-
 @Composable
-fun SignUp(
+fun SignUpScreen(
+    navController: NavHostController,
     modifier: Modifier = Modifier,
-    onSignUpSuccess: (email: String, password: String) -> Unit
+    onSignUpSuccess: (email: String, password: String) -> Unit,
+    viewModel: SignUpViewModel = viewModel(),
 ) {
-    var email by remember { mutableStateOf("") }
-    var password by remember { mutableStateOf("") }
+    val email by viewModel.email
```

Comment:
```suggestion
    val email by viewModel.email.collectAsStateWithLifecycle()
```
이렇게 작성하시는 것이 생명주기 측면에서 더 안전합니다. (그 외에 어떠한 장점이 있는지 찾아보시면 좋을 것 같아요)